src/modules/wholeBodyDynamics/main.cpp

```diff
@@ -360,10 +360,30 @@ class wholeBodyDynamics: public RFModule
         icub_type.head_version = 1;
         icub_type.legs_version = 1;
 
-        if (rf.check("headV2"))
+        // set icub_type.head_version int to 1 (default) or 2
+        if (rf.check("head_version"))
         {
-            yInfo("'headV2' option found. Using icubV2 head kinematics.\n");
-            icub_type.head_version = 2;
+            if (rf.find("head_version").asDouble()==1.0)
+            {
+                yInfo("\"head_version\" %g requested => Using V1 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+            }
+            else if (rf.find("head_version").asDouble()==2.0 ||
```

Comment:
I verified this parsing to work as expected both when the number is passed explicitly (`--head_version 1.0`), and when it is implicitly converted from int (`--head_version 1`). If anyone knows more robust ways to compare these command-line numbers, do let me know and I'll update the PR. :)